sw/device/tests/kmac_smoketest.c

```diff
@@ -151,10 +150,62 @@ const shake_test_t shake_tests[] = {
                    0x423323c9, 0xba7b8db4, 0x06c36eb0, 0x4fd75b36, 0xf0c70001,
                    0x0aefb1df, 0x6ae399e6, 0xf71930a6, 0xdef2206,  0x5ce2a640,
                    0x6a82fcf4, 0xa91b0815},
-        .digest_len = 102,  // Rate (r) is 34 words.
+        .digest_len = 3264 / 8 / 4,  // Rate (r) is 34 words.
     },
 };
 
+/**
+ * Run a SHA-3 test case with varying alignments.
+ */
+void run_sha3_alignment_test(dif_kmac_t *kmac) {
```

Comment:
We should either inline this OR move the other two loops into their own functions. Right now it is a bit inconsistent that this is the only test that has been pulled out of line.

Reply:
Probably makes sense to pull everything out of line now that there are more tests.